components/Modals/CharityManageModal.tsx

```diff
@@ -151,6 +153,83 @@ export default function CharityManageModal(props: Props) {
     if (props.onClose) props.onClose();
   };
 
+  const autoDistribute = () => {
+    const allKeys = Object.keys(percentMap);
+    const totalTouchedPercentage = allKeys
+      .filter((key) => touched[key])
+      .reduce((acc, key) => acc + percentMap[key], 0);
+  
+    const untouchedKeys = allKeys.filter((key) => !touched[key]);
+    const untouchedCount = untouchedKeys.length;
+  
+    if (allKeys.length === 1) {
+      // If there's only one recipient, set its percentage to 100%
+      percentMap[allKeys[0]] = 100;
+    } else if (totalTouchedPercentage > 100) {
+      // Scale down touched percentages proportionally to fit within 100%
+      const scalingFactor = 100 / totalTouchedPercentage;
+      allKeys.forEach((key) => {
+        if (touched[key]) {
+          percentMap[key] = Math.floor(percentMap[key] * scalingFactor);
+        }
+      });
+  
+      // Adjust for rounding issues
+      const scaledTotal = Object.values(percentMap).reduce((acc, cur) => acc + cur, 0);
+      const leftover = 100 - scaledTotal;
+      if (leftover > 0) {
+        percentMap[allKeys[0]] += leftover;
+      }
+    } else if (totalTouchedPercentage < 100) {
+      const remainingPercentage = 100 - totalTouchedPercentage;
+  
+      if (untouchedCount > 0) {
+        // Distribute the remaining percentage equally among untouched recipients
+        const equalDistribution = Math.floor(remainingPercentage / untouchedCount);
+        const distributedTotal = equalDistribution * untouchedCount;
+  
+        untouchedKeys.forEach((key) => {
+          percentMap[key] = equalDistribution;
+        });
+  
+        // Handle any leftover due to rounding
+        const leftover = remainingPercentage - distributedTotal;
+        if (leftover > 0) {
+          percentMap[untouchedKeys[0]] += leftover;
+        }
+      } else {
+        // If all recipients are touched, adjust the last one to fit 100%
```

Comment:
So in this case do you mean the last adjusted recipient should not get auto-adjusted/distributed ? by any means so far it makes it 100% ?

Reply:
No I mean it is auto-adjusting the wrong recipient

Reply:
Okay, based on what I understand, let me send a fix

Reply:
When there's two recipients, it should be

1. Untouched
2. Touched
Save --> auto-adjusts (1)

Then

1. Touched second
2. Touched first
Save --> auto-adjusts (2)

and then 

1. Touched second
2. Touched third (last)
Save --> auto-adjusts (1) (now it auto-adjusts 2 because it is the last in the array still)